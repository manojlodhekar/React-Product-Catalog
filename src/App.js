import React, { Component } from 'react';
import './App.css';
import './semantic/dist/semantic.min.css'
import ProductHeader from './products/ProductHeader';
import { Container, Button, Image } from 'semantic-ui-react';
import { Link } from 'react-router-dom';

const styles = {
  container: {
    alignItems: 'center',
    justifyContent: 'center',
    padding: 10,
    height: '100%',
    textAlign: 'center'
  },
}
class App extends Component {
  render() {
    return (
      <div>
        <div> 
          <Container textAlign='center' style={styles.container}>
            <ProductHeader subHeaderText={'Please click on Start butoon to start.'} />
            <Image centered size='large' src='/assets/wmt_h_r_c.jpg' />
            <Link to={"/products"} ><Button size='medium'>START</Button></Link>
          </Container>
          </div>
      </div>
    );
  }
}
export default App;
