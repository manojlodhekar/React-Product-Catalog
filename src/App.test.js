import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';

jest.mock('react-dom');

Enzyme.configure({ adapter: new Adapter() });

describe('App is defined', () => {
  it('should be defined', () => {
    expect(App).toBeDefined();
  });
  it('should render correctly', () => {
    const app = shallow(<App  />);
    expect(app).toHaveLength(1);
  });
  it('should render inner component', () => {
    const app = shallow(<App  />);
    expect(app).toMatchSnapshot();
  });

 });
