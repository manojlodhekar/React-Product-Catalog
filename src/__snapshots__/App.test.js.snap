// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App is defined should render inner component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <Container
                style={
                        Object {
                                "alignItems": "center",
                                "height": "100%",
                                "justifyContent": "center",
                                "padding": 10,
                                "textAlign": "center",
                              }
                }
                textAlign="center"
        >
                <ProductHeader
                        subHeaderText="Please click on Start butoon to start."
                />
                <Image
                        as="img"
                        centered={true}
                        size="large"
                        src="/assets/wmt_h_r_c.jpg"
                        ui={true}
                />
                <Link
                        replace={false}
                        to="/products"
                >
                        <Button
                                as="button"
                                size="medium"
                        >
                                START
                        </Button>
                </Link>
        </Container>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Container
          style={
                    Object {
                              "alignItems": "center",
                              "height": "100%",
                              "justifyContent": "center",
                              "padding": 10,
                              "textAlign": "center",
                            }
          }
          textAlign="center"
>
          <ProductHeader
                    subHeaderText="Please click on Start butoon to start."
          />
          <Image
                    as="img"
                    centered={true}
                    size="large"
                    src="/assets/wmt_h_r_c.jpg"
                    ui={true}
          />
          <Link
                    replace={false}
                    to="/products"
          >
                    <Button
                              as="button"
                              size="medium"
                    >
                              START
                    </Button>
          </Link>
</Container>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ProductHeader
              subHeaderText="Please click on Start butoon to start."
/>,
            <Image
              as="img"
              centered={true}
              size="large"
              src="/assets/wmt_h_r_c.jpg"
              ui={true}
/>,
            <Link
              replace={false}
              to="/products"
>
              <Button
                            as="button"
                            size="medium"
              >
                            START
              </Button>
</Link>,
          ],
          "style": Object {
            "alignItems": "center",
            "height": "100%",
            "justifyContent": "center",
            "padding": 10,
            "textAlign": "center",
          },
          "textAlign": "center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "subHeaderText": "Please click on Start butoon to start.",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "img",
              "centered": true,
              "size": "large",
              "src": "/assets/wmt_h_r_c.jpg",
              "ui": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Button
                as="button"
                size="medium"
>
                START
</Button>,
              "replace": false,
              "to": "/products",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": "button",
                "children": "START",
                "size": "medium",
              },
              "ref": null,
              "rendered": "START",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <Container
                    style={
                              Object {
                                        "alignItems": "center",
                                        "height": "100%",
                                        "justifyContent": "center",
                                        "padding": 10,
                                        "textAlign": "center",
                                      }
                    }
                    textAlign="center"
          >
                    <ProductHeader
                              subHeaderText="Please click on Start butoon to start."
                    />
                    <Image
                              as="img"
                              centered={true}
                              size="large"
                              src="/assets/wmt_h_r_c.jpg"
                              ui={true}
                    />
                    <Link
                              replace={false}
                              to="/products"
                    >
                              <Button
                                        as="button"
                                        size="medium"
                              >
                                        START
                              </Button>
                    </Link>
          </Container>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Container
            style={
                        Object {
                                    "alignItems": "center",
                                    "height": "100%",
                                    "justifyContent": "center",
                                    "padding": 10,
                                    "textAlign": "center",
                                  }
            }
            textAlign="center"
>
            <ProductHeader
                        subHeaderText="Please click on Start butoon to start."
            />
            <Image
                        as="img"
                        centered={true}
                        size="large"
                        src="/assets/wmt_h_r_c.jpg"
                        ui={true}
            />
            <Link
                        replace={false}
                        to="/products"
            >
                        <Button
                                    as="button"
                                    size="medium"
                        >
                                    START
                        </Button>
            </Link>
</Container>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <ProductHeader
                subHeaderText="Please click on Start butoon to start."
/>,
              <Image
                as="img"
                centered={true}
                size="large"
                src="/assets/wmt_h_r_c.jpg"
                ui={true}
/>,
              <Link
                replace={false}
                to="/products"
>
                <Button
                                as="button"
                                size="medium"
                >
                                START
                </Button>
</Link>,
            ],
            "style": Object {
              "alignItems": "center",
              "height": "100%",
              "justifyContent": "center",
              "padding": 10,
              "textAlign": "center",
            },
            "textAlign": "center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "subHeaderText": "Please click on Start butoon to start.",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": "img",
                "centered": true,
                "size": "large",
                "src": "/assets/wmt_h_r_c.jpg",
                "ui": true,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Button
                  as="button"
                  size="medium"
>
                  START
</Button>,
                "replace": false,
                "to": "/products",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "as": "button",
                  "children": "START",
                  "size": "medium",
                },
                "ref": null,
                "rendered": "START",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
